modules/OCPP201/device_model/composed_device_model_storage.cpp

```diff
@@ -6,92 +6,99 @@
 static constexpr auto VARIABLE_SOURCE_OCPP = "OCPP";
 
 namespace module::device_model {
-ComposedDeviceModelStorage::ComposedDeviceModelStorage(const std::string& libocpp_device_model_storage_address,
-                                                       const bool libocpp_initialize_device_model,
-                                                       const std::string& device_model_migration_path,
-                                                       const std::string& device_model_config_path) :
-    everest_device_model_storage(std::make_unique<EverestDeviceModelStorage>()),
-    libocpp_device_model_storage(std::make_unique<ocpp::v2::DeviceModelStorageSqlite>(
-        libocpp_device_model_storage_address, device_model_migration_path, device_model_config_path,
-        libocpp_initialize_device_model)),
-    device_model_map(get_device_model()) {
+ComposedDeviceModelStorage::ComposedDeviceModelStorage() {
+}
+
+bool ComposedDeviceModelStorage::register_device_model_storage(
+    std::string device_model_storage_id, std::unique_ptr<ocpp::v2::DeviceModelStorageInterface> device_model_storage) {
+    if (this->device_model_storages.find(device_model_storage_id) != this->device_model_storages.end()) {
+        return false;
+    }
+    this->device_model_storages[device_model_storage_id] = std::move(device_model_storage);
+    return true;
 }
 
 ocpp::v2::DeviceModelMap ComposedDeviceModelStorage::get_device_model() {
-    ocpp::v2::DeviceModelMap everest_dm = everest_device_model_storage->get_device_model();
-    ocpp::v2::DeviceModelMap libocpp_dm = libocpp_device_model_storage->get_device_model();
-    everest_dm.merge(libocpp_dm);
-    return everest_dm;
+    ocpp::v2::DeviceModelMap device_model_map;
+    for (const auto& [name, device_model_storage] : this->device_model_storages) {
+        device_model_map.merge(device_model_storage->get_device_model());
+    }
+    return device_model_map;
```

Comment:
```suggestion
    for (const auto& [name, device_model_storage] : this->device_model_storages) {
        this->device_model_map.merge(device_model_storage->get_device_model());
    }
    return this->device_model_map;
```

otherwise `get_variable_source()` later on doesn't have the whole merged device model